### ISSUE TITLE
source map url or presence change does not change asset md5 fingerprint

### ISSUE THREAD

**Issue by benhutton** [parcel-bundler/parcel#1454]
# 🐛 bug report

When I change the `--public-url` option, so that the `sourceMapUrl` at the bottom of my JS file changes, I should see the fingerprint of the file change...

When I use `--no-source-maps`, I should see the fingerprint of the file change...

Currently, neither change it.

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  "presets": [
    ["env", {
      "targets": {
        "browsers": "> 1%",
        "uglify": true
      },
      "useBuiltIns": true
    }]
  ],

  "plugins": [
    "syntax-dynamic-import",
    "transform-object-rest-spread",
    ["transform-class-properties", { "spec": true }]
  ]
}
```

## 🤔 Expected Behavior

When I change the `--public-url` option, so that the `sourceMapUrl` at the bottom of my JS file changes, I should see the fingerprint of the file change...

For example, I added `--public-url /dist` to my build command, so that my JS files end like `//# sourceMappingURL=/dist/application.b553e48e.map`. The filename should change now, as the contents of the file have changed.

When I add `--no-source-maps` to my build command, I should see the fingerprint of the file change...

## 😯 Current Behavior

When changing `--public-url` or using `--no-source-maps`, the md5 fingerprint does not change.

## 💁 Possible Solution

Include some of the options in the md5 calculation.

## 🔦 Context

I am trying to fix my source-map configuration, but the changes won't take effect until I make some other changes in my JS files, since my CDN and browsers are caching my assets very strongly.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.7.1 |
| Node             | 8.11.2 |
| npm/Yarn         | 5.6.0/1.2.1 |
| Operating System | Ubuntu 17.10 |

**Comment by DeMoorJasper** [comment]
Sourcemaps get added in packaging phase, therefore this doesn't change the output.
We could add some hacky way of doing this probably, but not sure if this is even a problem.

What about just adding a comment or tiny edit to your code to invalidate it instead? As this is a very niche use-case. (which should probably still get resolved eventually)

**Comment by mischnic** [comment]
Closely related to #1481

**Comment by devongovett** [comment]
For v2 I think this should be fixed by #5231.
